apps/examples/wifi_manager/wm_test/wm_test_connectbyrssi_test.c

```diff
@@ -0,0 +1,248 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <semaphore.h>
+#include <errno.h>
+#include <wifi_manager/wifi_manager.h>
+#include <netutils/netlib.h>
+#include "wm_test.h"
+#include "wm_test_log.h"
+#include "wm_test_utils.h"
+
+static sem_t g_wm_sem;
+static struct wt_options *opt;
+static char g_bssid[20];
+static char g_connected_ap_mac[20];
+
+#define TAG "[WTCR]"
+
+#define WM_TEST_SIGNAL              \
+	do {                            \
+		sem_post(&g_wm_sem);        \
+		WT_LOG(TAG, "send signal"); \
+	} while (0)
+
+#define WM_TEST_WAIT                \
+	do {                            \
+		WT_LOG(TAG, "wait signal"); \
+		sem_wait(&g_wm_sem);       \
+	} while (0)
+
+
+static void _wt_sta_connected(wifi_manager_cb_msg_s msg, void *arg);
+static void _wt_sta_disconnected(wifi_manager_cb_msg_s msg, void *arg);
+static void _wt_scan_done(wifi_manager_cb_msg_s msg, void *arg);
+
+static wifi_manager_cb_s g_wifi_callbacks = {
+	_wt_sta_connected,
+	_wt_sta_disconnected,
+	NULL,
+	NULL,
+	_wt_scan_done,
+};
+
+void _wt_sta_connected(wifi_manager_cb_msg_s msg, void *arg)
+{
+	WT_LOG(TAG, "-->res(%d)", msg.res);
+	WT_LOG(TAG, "bssid %02x:%02x:%02x:%02x:%02x:%02x",
+		msg.bssid[0], msg.bssid[1],
+		msg.bssid[2], msg.bssid[3],
+		msg.bssid[4], msg.bssid[5]);
+
+	if (WIFI_MANAGER_SUCCESS == msg.res) {
+		WT_LOG(TAG, "Station Connected");
+		snprintf(g_connected_ap_mac, sizeof(g_connected_ap_mac), "%02x:%02x:%02x:%02x:%02x:%02x",
+			msg.bssid[0], msg.bssid[1],
+			msg.bssid[2], msg.bssid[3],
+			msg.bssid[4], msg.bssid[5]);
+	} else {
+		WT_LOGE(TAG, "Station Not Connected");
+	}
+	WM_TEST_SIGNAL;
+}
+
+void _wt_sta_disconnected(wifi_manager_cb_msg_s msg, void *arg)
+{
+	WT_LOG(TAG, "--> res(%d) reason %d", msg.res, msg.reason);
+	WM_TEST_SIGNAL;
+}
+
+void _wt_scan_done(wifi_manager_cb_msg_s msg, void *arg)
+{
+	WT_LOG(TAG, "-->");
+	/* Make sure you copy the scan results onto a local data structure.
+	 * It will be deleted as soon as you exit this function.
+	 */
+	if (msg.res != WIFI_MANAGER_SUCCESS || msg.scanlist == NULL) {
+		WM_TEST_SIGNAL;
+		return;
+	}
+	wt_print_scanlist(msg.scanlist);
+	wifi_manager_scan_info_s *wifi_scan_iter = msg.scanlist;
+	while (wifi_scan_iter != NULL) {
+		if (strncmp(wifi_scan_iter->ssid, opt->ssid, WIFIMGR_SSID_LEN) == 0) {
+			strncpy(g_bssid, wifi_scan_iter->bssid, strlen(wifi_scan_iter->bssid));
```

Comment:
Please add the logic to find the bssid for highest RSSI value

Reply:
From underlying wifi driver callback, scanlist will always contain sorted scanned data as per best rssi value. 
So for a given ssid, best rssi value will come first, which is being copied in g_bssid followed by loop termination. 
I will modify my code to make it independent from ordering of scanned data from underlaying wifi driver callback.